Translate uncontrolled components doc (#103)
reactjs/de.reactjs.org: content/docs/uncontrolled-components.md

---
id: uncontrolled-components
title: Unkontrollierte Komponenten
permalink: docs/uncontrolled-components.html
---

> In den meisten Fällen empfehlen wir die Verwendung von [kontrollierten Komponenten](/docs/forms.html) zur Implementierung von Formularen. In einer kontrollierten Komponente werden Formulardaten von einer React Komponente verarbeitet. Die Alternative sind unkontrollierte Komponenten, bei denen die Formulardaten von dem DOM selbst verarbeitet werden.

Reviewer: ```suggestion
In den meisten Fällen empfehlen wir die Verwendung von [kontrollierten Komponenten](/docs/forms.html) zur Implementierung von Formularen. In einer kontrollierten Komponente werden Formulardaten von einer React Komponente verarbeitet. Die Alternative sind unkontrollierte Komponenten, bei denen die Formulardaten vom DOM selbst verarbeitet werden.
```

> Um eine unkontrollierte Komponente zu erstellen, kannst du anstelle eines Event-Handler für jede Statusaktualisierung [ein ref](/docs/refs-and-the-dom.html) verwenden, um Formularwerte aus dem DOM abzurufen.

Reviewer: ```suggestion
Um eine unkontrollierte Komponente zu erstellen, kannst du anstelle eines Event-Handlers für jede Aktualisierung des States [ein ref](/docs/refs-and-the-dom.html) verwenden, um Formularwerte aus dem DOM abzurufen.
```

}
```

[**Auf CodePen ausprobieren**](https://codepen.io/gaearon/pen/WooRWa?editors=0010)

Mit unkontrollierten Komponenten ist es manchmal einfacher, React und nicht-React Code zu integrieren, da die Quelle der Wahrheit in einer unkontrollierten Komponente das DOM ist. Es kann auch etwas weniger Code sein, wenn du eine schnelle Lösung haben möchtest. Ansonsten solltest du in der Regel kontrollierte Komponenten verwenden.

> Wenn es immer noch nicht klar ist, welche Art von Komponente du für eine bestimmte Situation verwenden solltest, findest du möglicherweise [diesen Artikel über kontrollierte versus unkontrollierte Komponente](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) hilfreich.

Reviewer: ```suggestion
Wenn es immer noch nicht klar ist, welche Art von Komponente du für eine bestimmte Situation verwenden solltest, findest du möglicherweise [diesen Artikel über kontrollierte gegen unkontrollierte Komponente](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) hilfreich.
```

}
```

> Ebenso understützen `<input type="checkbox">` und `<input type="radio">` `defaultChecked`, und `<select>` und `<textarea>` understützen `defaultValue`.

Reviewer: ```suggestion
Ebenso unterstützen `<input type="checkbox">` und `<input type="radio">` `defaultChecked`, und `<select>` und `<textarea>` unterstützen `defaultValue`.
```

## Das Datei input Element {#the-file-input-tag}

In HTML kann der Benutzer mit `<input type="file">` eine oder mehrere Dateien aus dem Gerätespeicher auswählen, die auf einen Server hochgeladen oder von JavaScript über die [File API](https://developer.mozilla.org/de/docs/Web/API/File/Zugriff_auf_Dateien_von_Webapplikationen) bearbeitet werden können.

```html
<input type="file" />
```

> In React, ein `<input type="file" />` ist immer eine unkontrollierte Komponente, da ihr Wert nur von einem Benutzer und nicht programmatisch festgelegt werden kann.

Reviewer: ```suggestion
In React ist `<input type="file" />` immer eine unkontrollierte Komponente, da ihr Wert nur von einem Benutzer und nicht programmatisch festgelegt werden kann.
```
